hotels-backend/Hotels/Controllers/HotelsController.cs

```diff
@@ -0,0 +1,129 @@
+using Microsoft.EntityFrameworkCore;
+using Hotels.Models;
+
+namespace Hotels.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class HotelsController : ControllerBase
+    {
+        private readonly HotelsDBContext _context;
+
+        public HotelsController(HotelsDBContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Hotels
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Hotel>>> GetHotels()
+        {
+            return await _context.Hotels.ToListAsync();
+        }
+
+        // GET: api/Hotels/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Hotel>> GetHotel(int id)
+        {
+            var hotel = await _context.Hotels.FindAsync(id);
+
+            if (hotel == null)
+            {
+                return NotFound();
+            }
+
+            return hotel;
+        }
+
+        // PUT: api/Hotels/5
+        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutHotel(int id, Hotel hotel)
+        {
+            if (id != hotel.Id)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(hotel).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!HotelExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return NoContent();
+        }
+
+        // POST: api/Hotels
+        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        [HttpPost]
+        public async Task<ActionResult<Hotel>> PostHotel(Hotel hotel)
+        {
+            _context.Hotels.Add(hotel);
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateException)
+            {
+                if (HotelExists(hotel.Id))
+                {
+                    return Conflict();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return CreatedAtAction("GetHotel", new { id = hotel.Id }, hotel);
+        }
+
+        // DELETE: api/Hotels/5
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> DeleteHotel(int id)
+        {
+            var hotel = await _context.Hotels.FindAsync(id);
+            if (hotel == null)
+            {
+                return NotFound();
+            }
+
+            _context.Hotels.Remove(hotel);
+            var rooms = await _context.Rooms.ToListAsync();
+            rooms = rooms.Where(x => x.HotelId == id).ToList();
```

Comment:
In case when several method calls are chained start each method call from a new line.
```suggestion
            rooms = rooms
                 .Where(x => x.HotelId == id)
                 .ToList();
```